examples/generative/corrdiff/train.py

```diff
@@ -303,57 +327,46 @@ def main(cfg: DictConfig) -> None:
                             "validation_loss", average_valid_loss, cur_nimg
                         )
 
-        cur_nimg += cfg.training.hp.total_batch_size
-        done = cur_nimg >= cfg.training.hp.training_duration
-        if (not done) and (
-            cur_nimg < tick_start_nimg + cfg.training.io.print_progress_freq
-        ):  # TODO revert
-            continue
-
-        # Print stats
-        tick_end_time = time.time()
-        if dist.rank == 0:
-            fields = []
-            fields += [f"tick {cur_tick:<5d}"]
-            fields += [f"samples {(cur_nimg):<9.1f}"]
-            fields += [f"training_loss {average_loss:<7.2f}"]
-            fields += [f"learning_rate {current_lr:<7.8f}"]
-            fields += [f"total_sec {(tick_end_time - start_time):<7.1f}"]
-            fields += [f"sec_per_tick {(tick_end_time - tick_start_time):<7.1f}"]
-            fields += [
-                f"sec_per_sample {((tick_end_time - tick_start_time) / (cur_nimg - tick_start_nimg)):<7.2f}"
-            ]
-            fields += [
-                f"cpu_mem_gb {(psutil.Process(os.getpid()).memory_info().rss / 2**30):<6.2f}"
-            ]
-            fields += [
-                f"peak_gpu_mem_gb {(torch.cuda.max_memory_allocated(dist.device) / 2**30):<6.2f}"
-            ]
-            fields += [
-                f"peak_gpu_mem_reserved_gb {(torch.cuda.max_memory_reserved(dist.device) / 2**30):<6.2f}"
-            ]
-            logger0.info(" ".join(fields))
-        torch.cuda.reset_peak_memory_stats()
+        if is_time_for_periodic_task(
+            cfg.training.io.print_progress_freq, rank_0_only=True
+        ):
+            # Print stats if we crossed the printing threshold with this batch
+            tick_end_time = time.time()
+            if dist.rank == 0:
```

Comment:
I don't think this is needed as you have already set `rank_0_only=True`

Reply:
Removed in https://github.com/NVIDIA/modulus/pull/636/commits/3df9d26f94ec5d13075e1bd6b1f919a9b07b114c.